### ISSUE TITLE
returning data from stream in flutter

### ISSUE THREAD

**Issue by yutech23** [supabase-community/supabase-flutter#218]
doesn't work listening to a specific column.

My Table;
`database columns(category5) = (category5_id,name,fk_category4_id)`

My code is;
```dart
db.supabase
.from('category5:fk_category4_id=eq.${selectCategory5!.keys.first}')
.stream(['category5_id,name']).execute();

return data => all columns.
```

**Comment by dshukertjr** [comment]
@yutech23 
Thanks for opening this issue.

Are you unable to receive all insert, update, and delete events, or is it only delete events? 

There was an issue on our realtime server where the delete events stopped coming through when listening with eq filter, which is resolved now.

**Comment by iampopal** [comment]
@dshukertjr 
In my case 
The stream return for the first time, After initializing str again 
Stream failed to load

**Comment by iampopal** [comment]
We need to able to cancel stream with StreamSubscription 
And 
Initialize again 

Like 
If user not have internet, an error throws form stream, 
Then 
When user get internet and press refresh we shall able to initialize stream again

**Comment by dshukertjr** [comment]
@iampopal That sounds like a separate issue, so would you want to open a new issue and discuss it there? It would be helpful if you could follow the issue template and include some reproducible code as well!

**Comment by dshukertjr** [comment]
@yutech23 The issue with not being able to receive delete events has been resolved, so I would like to close this one assuming you were facing the same issue, but if you are still experiencing the issue that you open this thread for, please feel free to reopen it at any time!

**Comment by yutech23** [comment]
> @yutech23 The issue with not being able to receive delete events has been resolved, so I would like to close this one assuming you were facing the same issue, but if you are still experiencing the issue that you open this thread for, please feel free to reopen it at any time!

Hi  Sorry for late reply, this problem continues. 
![image](https://user-images.githubusercontent.com/42283661/194079070-5d8876b1-20e4-4122-89d2-1966d503417a.png)

return value:

![image](https://user-images.githubusercontent.com/42283661/194079260-9f23cd52-9d6c-43bc-a6b3-5273ffa34769.png)

**Comment by yutech23** [comment]
> @yutech23 Thanks for notifying that it is still not working.
> 
> May I ask how you are listening to your stream? Also, you have a provided an screenshot of a log, but how are you logging the data? If you could provide the code for it as well, it would be amazing!

Of course.

Listining
![image](https://user-images.githubusercontent.com/42283661/194597204-1444322e-624c-40bd-8cc4-ed1a6c8ba6bc.png)

like this Data input;

![image](https://user-images.githubusercontent.com/42283661/194598280-24a961af-3b57-42ea-98e0-47dda0b8e52d.png)

**Comment by dshukertjr** [comment]
@yutech23 
Thanks for providing the code. Generally, it would be more awesome if you could copy and paste the code rather than taking a screenshot of it!

 I want to make sure that I understand the issue here, but in the title of this issue, you wrote that relatime does not work when listening to a specific column, but it looks like you are not listening to a specific column in [this example](https://github.com/supabase-community/supabase-flutter/issues/218#issuecomment-1268477168). Is the issue that you are not able to receive any realtime data at all?

If so, there are few things that come to my mind here. 
- Persist the stream data as state. More details [here](https://supabase.com/docs/reference/dart/next/stream#using-stream-with-streambuilder). In your case, I would just make the stream a property of the bloc instead of returning it in a method.
- Make sure you have realtime turned on. You can turn it on by managing replication under database menu of your Supabase dashboard.

**Comment by yutech23** [comment]
Hi,
I made an example similar to this problem. 
My database column names are ('category1_id','name')
I want to stream only 'name' column.

My Code;
ElevatedButton(
                      onPressed:() async{
                        var res =  db.supabase.from('category1').stream(['name']).execute();
                        res.listen((event) {
                          print(event);
                        });

                      },
                      child: Text('Gönder'))

Returning Data;
[{category1_id: 15, name: Unisex}, {category1_id: 23, name: Kadın}, {category1_id: 25, name: Erkek}]

Note: Returned All Table data.

Thank you for all of replies.

**Comment by Vinzent03** [comment]
@yutech23 I don't think you can stream specific columns. Is `name` your primary key column?

**Comment by dshukertjr** [comment]
@yutech23 
Yup, with `stream()`, there is currently no way to limit which columns to retrieve. It can only retrieve all the columns in a table. The `List` you pass inside the `.stream()` is list of primary keys for the table.

@Vinzent03 
I wonder if we can make the API better by making the primary keys named parameters like this:

```dart
supabase.from('category1').stream(primaryKeys: ['id'])
```

This might open up the api to allow only retrieving certain columns like this:
```dart
supabase.from('category1').stream(primaryKeys: ['id'], columns: ['name', 'category'])
```

**Comment by Vinzent03** [comment]
I think that makes it more clear. Do you want to support the column select on client or just prepare for server select with named parameters. But we should change the name, because there is only one primary key, but with multiple columns.

**Comment by dshukertjr** [comment]
@Vinzent03 
>Do you want to support the column select on client or just prepare for server select with named parameters.

Sorry, didn't catch this one. What do you mean by "server select"?

>But we should change the name, because there is only one primary key, but with multiple columns.

Oh yeah, makes sense!

**Comment by Vinzent03** [comment]
@dshukertjr I mean whether you want to prepare for a future feature of the realtime server of selecting columns, because that's currently not possible, right? Or you want to select columns in the local dart code? So basically where the column filtering happens, on realtime server or local.

**Comment by dshukertjr** [comment]
@Vinzent03 
Thanks for the clarification! I was thinking about this over the weekend, and was wondering if we really need to add the ability to restrict which columns to select. At least since this would be a new feature, we don't have to implement it right now. Let's wait on this one for now, and just go with changing the syntax around primaryKey for now!

**Comment by Vinzent03** [comment]
@dshukertjr lets include the stream parameter change in supabase-dart 1.0.0, right?

**Comment by yutech23** [comment]
> I don't think you can stream specific columns

 'name' is not primary key column.
category1_id is primary key.

**Comment by dshukertjr** [comment]
@yutech23 
Yup, in your case, you need to get the stream like this:
```dart
db.supabase.from('category1').stream(['category1_id']).execute();
```

Note that there is no way to limit which columns to retrieve at the moment, and `.stream()` will return all the columns for the table.

**Comment by Vinzent03** [comment]
@yutech23  Is your original issue solved, so we can close this? Selecting columns in stream should be discussed in another issue.

**Comment by yutech23** [comment]
> @yutech23 Is your original issue solved, so we can close this? Selecting columns in stream should be discussed in another issue.

Of course you could close.
